src/Orleans.Runtime/Core/InsideRuntimeClient.cs

```diff
@@ -411,6 +411,7 @@ public void SniffIncomingMessage(Message message)
 
                     if (message.Direction != Message.Directions.OneWay)
                     {
+                        TransactionContext.Clear();
```

Comment:
shouldn't this be cleared regardless of the direction?

Reply:
It is, but it probably doesn't matter.  What's critical here is whether the context is still set when building the response message.  For calls that created the context, or when exceptions are thrown, the transaction context should be cleared prior to building a response.